### PR TITLE
IRISHUB-317,318: realize the depositProcedureParameter and its UnitTest

### PR DESCRIPTION
**Pull request by MrXJC**

### CODE REVIEW

File: modules/gov/gov_params.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+var DepositProcedureParameter DepositProcedureParam
+
+type DepositProcedureParam struct {
+	Value DepositProcedure
+	ps    params.Setter
+	pg    params.Getter
+}
+
+func (param *DepositProcedureParam) InitGenesis(genesisState interface{}) {
+	if value, ok := genesisState.(DepositProcedure); ok {
```
Comment (wukongcheng) [inline modules/gov/gov_params.go]: What happens if the genesisState is nil？

File: modules/gov/gov_params.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
+)
+
+var DepositProcedureParameter DepositProcedureParam
+
+type DepositProcedureParam struct {
+	Value DepositProcedure
+	ps    params.Setter
+	pg    params.Getter
```
Comment (wukongcheng) [inline modules/gov/gov_params.go]: psetter & pgetter will be more readable

File: modules/gov/gov_params.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	return true
+}
+
+func (param *DepositProcedureParam) ToJson() string {
+	jsonBytes, _ := json.Marshal(param.Value)
+	return string(jsonBytes)
+}
+
+func (param *DepositProcedureParam) Update(ctx sdk.Context, jsonStr string) {
+	if err := json.Unmarshal([]byte(jsonStr), &param.Value); err == nil {
```
Comment (wukongcheng) [inline modules/gov/gov_params.go]: Should return error if Unmarshal failed?

File: modules/gov/gov_params.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+func (param *DepositProcedureParam) Valid(jsonStr string) sdk.Error {
+
+	var err error
+
+	if err = json.Unmarshal([]byte(jsonStr), &param.Value); err == nil {
+
+		if param.Value.MinDeposit[0].Denom != "iris" {
+			return sdk.NewError(DefaultCodespace, 102, fmt.Sprintf("It should be iris "))
```
Comment (wukongcheng) [inline modules/gov/gov_params.go]: Should use the enum instead of the magic number 102

File: modules/gov/msgs_test.go [hunk 5]
```diff
@@ -39,8 +39,10 @@ func TestMsgSubmitProposal(t *testing.T) {
 		{"Test Proposal", "the purpose of this proposal is to test", ProposalTypeText, addrs[0], coinsMulti, true},
 	}
 
+	var params Params
```
Comment (wukongcheng) [inline modules/gov/msgs_test.go]: We should create a new folder "params" under the "gov", and put all these source files in it

File: modules/gov/gov_params.go (hunk 1 above)
Comment (MrXJC) [reply modules/gov/gov_params.go]: You can see the UnitTest. The value of `ok` is `false`.

File: modules/gov/gov_params.go (hunk 3 above)
Comment (MrXJC) [reply modules/gov/gov_params.go]: Function `Valid` have done !

File: modules/gov/gov_params.go (hunk 4 above)
Comment (MrXJC) [reply modules/gov/gov_params.go]: It maybe need more design.

File: modules/gov/msgs_test.go (hunk 5 above)
Comment (MrXJC) [reply modules/gov/msgs_test.go]: Ok, I will do it tomorrow.

**Pull request merged**